examples/bzlmod_build_file_generation/MODULE.bazel

```
@@ -68,13 +65,19 @@ python.toolchain(
 # are not catenated for ease of reading.
 use_repo(python, PYTHON_NAME)
 use_repo(python, "python3_toolchains")
-use_repo(python, PYTHON_INTERPRETER)
 
 # Register an already-defined toolchain so that Bazel can use it during toolchain resolution.
 register_toolchains(
     "@python3_toolchains//:all",
 )
 
+pypi = use_extension("@rules_python//python:interpreter_extensions.bzl", "pypi_ext")
+pypi.install(
+    name = "interpreter_python3",
+    python_name = PYTHON_NAME,
+)
+use_repo(pypi, "interpreter_python3")
```

`pypi` stands for Python Package Index, so using it as a name may be confusing.

```suggestion
interpreter = use_extension("@rules_python//python:interpreter_extensions.bzl", "pypi_ext")
interpreter.install(
    name = "interpreter_python3",
    python_name = PYTHON_NAME,
)
use_repo(interpreter, "interpreter_python3")
```

---

Lets call it "host_interpreter" instead of just "interpreter". This is to make it clear its for the host system (i.e. repo rules) and not general usage.

---

`host_interpreter` sounds like the interpreter installed on the host, not the interpreter the rules downloaded.  Any other options?

---

I'm fine with "interpreter". The only other name I can think of is "toolchain_interpreter".